controllers/awsadapterconfig_controller.go

```diff
@@ -105,22 +105,37 @@ func (r *AWSAdapterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 	clusterFound := false
 
-	if callerIdentity, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{}); err == nil && callerIdentity.Account != nil {
+	callerIdentity, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
+	if err != nil || callerIdentity.Account == nil {
+		if callerIdentity.Account == nil {
+			err = fmt.Errorf("callerIdentity nil")
+		}
+
+		msg := "error occurred while fetching account id"
+		l.Error(err, msg)
+		return r.updateLastPollStatusFailure(ctx, objOld, msg, err, &l, time.Now())
+	} else {
```

Comment:
The original return-early suggestion was to avoid this else altogether, and reduce the nesting.  Since the if was definitely going to return.

Reply:
On second thoughts, it is ok to keep it in the if-else, will be easier to move out to separate functions later.